src/main/java/TaskList.java

```diff
@@ -0,0 +1,192 @@
+            throw new TulipTaskException.InvalidTaskDescriptionException();
+        }
+
+        if (from == null) {
+            throw new TulipTaskException.InvalidStartDateException();
+        }
+
+        if (to == null) {
+            throw new TulipTaskException.InvalidEndDateException();
+        }
+
+        Event task = new Event(argument, from, to);
+        addTask(task);
+    }
+
+    public void getTaskIndex(String indexString) {
+        int index = Integer.parseInt(indexString) - 1;
+
+        if (index < 0 || index > taskList.size() - 1) {
+            this.taskIndex = INVALID_INDEX;
+            return;
+        }
+
+        this.taskIndex = index;
+    }
+
+    public void markTaskAsDone(HashMap<String, String> commandArguments) throws TulipTaskException.InvalidTaskIndexException {
+        String argument = commandArguments.get("argument");
+        getTaskIndex(argument);
+
+        if (this.taskIndex == INVALID_INDEX) {
+            throw new TulipTaskException.InvalidTaskIndexException();
+        }
+
+        Task task = taskList.get(this.taskIndex);
+        task.markAsDone();
+        System.out.println("--------------------------------------------");
+        System.out.println("Great job! I have marked this task as done: ");
+        System.out.println(task);
+        System.out.println("--------------------------------------------");
+    }
+
+    public void markTaskAsNotDone(HashMap<String, String> commandArguments) throws TulipTaskException.InvalidTaskIndexException {
+        String argument = commandArguments.get("argument");
+        getTaskIndex(argument);
+
+        if (this.taskIndex == INVALID_INDEX) {
+            throw new TulipTaskException.InvalidTaskIndexException();
+        }
+
+        Task task = taskList.get(this.taskIndex);
+        task.markAsNotDone();
+        System.out.println("--------------------------------------------");
+        System.out.println("Okay, I have marked this task as not done: ");
+        System.out.println(task);
+        System.out.println("--------------------------------------------");
+    }
+
+    public void saveTaskToFile() {
+        System.out.println("--------------------------------------------");
+        SaveTaskList.saveTasks(this.taskList);
+        System.out.println("--------------------------------------------");
+    }
+
+    public void loadTaskFromFile() {
+        List<String> list = SaveTaskList.loadTasks();
+        for (int i = 0; i < Objects.requireNonNull(list).size(); i++) {
+            parseTask(list.get(i));
+        }
+        System.out.println("--------------------------------------------");
+        System.out.println("Tasks have been successfully loaded!");
+        System.out.println("--------------------------------------------");
+    }
+
+    public void parseTask(String line) {
+        String type = line.substring(1, 2);
+        boolean completed = line.charAt(4) == 'X';
+
+        switch (type) {
```

Comment:
For your switch statements, maybe you could follow the style mentioned [here](https://se-education.org/guides/conventions/java/basic.html#layout).